src/Xamarin.Android.Build.Tasks/Tasks/ProcessAssemblies.cs

```diff
@@ -46,82 +50,118 @@ public override bool RunTask ()
 				}
 			}
 
+			// We only need to "dedup" assemblies when there is more than one RID
+			if (RuntimeIdentifiers.Length > 1) {
+				Log.LogDebugMessage ("Deduplicating assemblies per RuntimeIdentifier");
+				DeduplicateAssemblies (output, symbols);
+			} else {
+				Log.LogDebugMessage ("Found a single RuntimeIdentifier");
+				SetMetadataForAssemblies (output, symbols);
+			}
+
+			OutputAssemblies = output.ToArray ();
+			ResolvedSymbols = symbols.Values.ToArray ();
+
+			// Set ShrunkAssemblies for _RemoveRegisterAttribute and <BuildApk/>
+			if (PublishTrimmed) {
+				ShrunkAssemblies = OutputAssemblies.Select (a => {
```

Comment:
Do we want to use Linq here? We usually try to avoid it for performance reasons.

Reply:
I changed this one to a `foreach` loop.

There is another place in here that uses `.Where(..).GroupBy(...)`, which I think is a good usage of Linq.